Lib/test/test_ctypes/test_unions.py

```diff
@@ -0,0 +1,32 @@
+import ctypes
+import sys
+import unittest
+import warnings
+from ctypes import Union
+
+
+UnionType = type(Union)
+# _CData is not exported to Python, so we have to access it from __base__.
+_CData = Union.__base__
+
+
+class ArrayTestCase(unittest.TestCase):
+    def test_inheritance_hierarchy(self):
+        self.assertEqual(_CData.__name__, "_CData")
+        self.assertEqual(Union.mro(), [Union, _CData, object])
+
+        self.assertEqual(UnionType.__name__, "UnionType")
+        self.assertEqual(type(UnionType), type)
+
+    def test_instantiation(self):
+        with self.assertRaisesRegex(TypeError, "abstract class"):
+            Union()
+
+        UnionType("Foo", (Union,), {})
+
+    def test_immutability(self):
+        Union.foo = "bar"
```

Comment:
To my superise, `Union` is mutable now.

Reply:
Was it not in earlier Python versions?

Reply:
Yes, testing on 3.10, it's also mutable. I haven't dug too much into it, maybe it's a bug, but I think for now we should keep the behavior?

Reply:
Yes, we aim to keep the current behaviour.